### PR TITLE
Replace pytokenizations with internal alignment

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

The internal alignment algorithm requires the concatenation of the two tokens lists to be identical with the following exceptions:

* Differing capitalization is allowed
* Differing whitespace within tokens containing non-whitespace is allowed
* Differing leading and trailing whitespace tokens are allowed as long there is a potential alignment on token boundaries
* Differing internal whitespace-only tokens are not permitted

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/training/align.py [hunk 1]
```diff
@@ -32,3 +30,78 @@ def _make_ragged(indices):
...
+    tokens_a = tokens_a[:trailing_ws_start_a] + trailing_ws_a
+    tokens_b = tokens_b[:trailing_ws_start_b] + trailing_ws_b
+    # Check that the strings align, allowing only differences in capitalization
+    if "".join(tokens_a).lower() != "".join(tokens_b).lower():
+        raise ValueError(Errors.E949)
+    # Store the set of aligned token indices per token
+    a2b = [set() for _ in range(len(tokens_a))]
+    b2a = [set() for _ in range(len(tokens_b))]
+    # Create iterators for token indices per character position
+    token_by_char_a = chain(*[[i] * len(x) for i, x in enumerate(tokens_a)])
```
Comment (honnibal) [inline spacy/training/align.py:76]: Is this fast enough? It looks terrible.

Comment (adrianeboyd) [reply spacy/training/align.py:76]: It does look terrible. I'll refactor it a bit.

**Comment by adrianeboyd** [comment]
The earlier version was ~2-3x slower than pytokenizations and the refactored version is ~1.1x slower? It depends a bit on how misaligned the texts are and how long the tokens are.

**Comment by adrianeboyd** [comment]
Let me look at the whitespace handling again...

**Pull request closed without merging**